runtime/src/main/java/org/corfudb/runtime/clients/NettyClientRouter.java

```diff
@@ -411,49 +398,59 @@ public void stop(boolean shutdown) {
      *     or a timeout in the case there is no response.
      */
     public <T> CompletableFuture<T> sendMessageAndGetCompletable(ChannelHandlerContext ctx,
-        CorfuMsg message) {
+        @NonNull CorfuMsg message) {
         boolean isEnabled = MetricsUtils.isMetricsCollectionEnabled();
-        // Check the connection future, and wait for it to be successful before
-        // sending the message.
+
+        // Check the connection future. If connected, continue with sending the message.
+        // If timed out, return a exceptionally completed with the timeout.
         try {
             connectionFuture
                 .get(parameters.getConnectionTimeout().toMillis(), TimeUnit.MILLISECONDS);
         } catch (InterruptedException e) {
             Thread.currentThread().interrupt();
             throw new UnrecoverableCorfuInterruptedError(e);
         } catch (TimeoutException | ExecutionException e) {
-            // If we timed out, return a CompletableFuture exceptionally completed
-            // with the timeout.
             CompletableFuture<T> f = new CompletableFuture<>();
             f.completeExceptionally(e);
             return f;
         }
-        final Timer.Context context = MetricsUtils
-            .getConditionalContext(isEnabled, timerSyncOp);
+
+        // Set up the timer and context to measure request
+        final String timerName = CorfuComponent.CR.toString() + message.getMsgType().name().toLowerCase();
```

Comment:
I'm somewhat concerned about common case performance. It seems that this string construction occurs on every message round trip, even if the timing is off. Is there a way to avoid this?

Reply:
Done. (Adding a map to avoid string construction.)